packages/webdriverio/src/utils/actions/key.ts

```diff
@@ -20,7 +20,7 @@ export default class KeyAction extends BaseAction {
             // check capabilities first
             platformName && platformName.match(/mac(\s)*os/i) ||
             // if not set, expect we run locally
-            os.type().match(/darwin/i)
+            (this.instance.options.hostname?.match(/local/i) && os.type().match(/darwin/i))
```

Comment:
`hostname` could also be `127.0.0.1`. I think it is impossible to really know whether the session runs on ones machine or not. So let's just check for `localhost` and `127.0.0.1`